common/src/types/v0/store/replica.rs

```diff
@@ -119,10 +207,20 @@ impl ResourceUuid for ReplicaSpec {
 
 impl From<ReplicaSpec> for models::ReplicaSpec {
     fn from(src: ReplicaSpec) -> Self {
-        Self::new(
+        let pool_id = match src.pool.clone() {
```

Comment:
We have a lot of this code repetition here, we should add methods to PoolRef, example:
fn name(&self) -> PoolId { ... }
fn uuid(&self) -> Option<PoolUuid> { ... }